### PR TITLE
Port to meshery-kit and meshery-adapter-library

### LINKED ISSUE #65: Replace meshery-osm adapter to use meshey-kit and meshery-framework as primary dependencies
**Enhancement Description**
A brief description of the enhancement.

As we discussed we should replace `meshery-osm` adapter to use `meshey-kit `and `meshery-framework` as primary dependencies. So I'm going to work on `meshery-osm` adapter @kumarabd @leecalcote @mgfeller 

**Related Challenges**
A brief description of what the problem is. (e.g. I need to be able to...)

---
[Optional] **Alternatives**
A brief description of any alternative solutions or features you've considered.

[Optional] **Additional context**

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: aisuko <urakiny@gmail.com>

**Description**

This PR fixes #65

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Aisuko** [comment]
I saw the result of the CI, but I'm not sure what's the problems.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (leecalcote) [inline Makefile:1]: @kushthedude the workflow for each adapter is consistent. Is the Makefile for each adapter consistent as well?

Comment (kushthedude) [reply Makefile:1]: Workflow for each adapter is not consistent, only build and release flows are consistent for each adapter.

**Review by mgfeller — CHANGES_REQUESTED** [review]
internal/config appears to be missing

File: go.mod [hunk 2]
```diff
@@ -1,21 +1,12 @@
...
-	github.com/gophercloud/gophercloud v0.4.0 // indirect
-	github.com/layer5io/gokit v0.1.16
-	github.com/pkg/errors v0.9.1
-	github.com/sirupsen/logrus v1.6.0
-	golang.org/x/net v0.0.0-20200707034311-ab3426394381
-	google.golang.org/grpc v1.31.0
-	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/client-go v0.18.8
+	github.com/layer5io/meshery-adapter-library v0.1.7
+	github.com/layer5io/meshkit v0.1.27
```
[the remark this answers is not in the document]
Comment (kumarabd) [reply go.mod]: Done

File: go.mod [hunk 3]
```diff
@@ -1,21 +1,12 @@
...
-	github.com/golang/protobuf v1.4.2
-	github.com/gophercloud/gophercloud v0.4.0 // indirect
-	github.com/layer5io/gokit v0.1.16
-	github.com/pkg/errors v0.9.1
-	github.com/sirupsen/logrus v1.6.0
-	golang.org/x/net v0.0.0-20200707034311-ab3426394381
-	google.golang.org/grpc v1.31.0
-	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/client-go v0.18.8
+	github.com/layer5io/meshery-adapter-library v0.1.7
```
[the remark this answers is not in the document]
Comment (kumarabd) [reply go.mod]: Done

File: osm/install.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	err = cmd.Wait()
+	if err != nil {
+		return st, err
+	}
+
+	return status.Removed, nil
+}
+
+// TODO: Should add the function into meshkit
```
[the remark this answers is not in the document]
Comment (kumarabd) [reply osm/install.go]: Yep, it came in with previous changes, removed now.

File: main.go [hunk 5]
```diff
@@ -15,49 +15,61 @@
...
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/grpclog"
-
-	"github.com/sirupsen/logrus"
-
-	mesh "github.com/layer5io/meshery-osm/meshes"
+	"github.com/layer5io/meshery-adapter-library/adapter"
+	"github.com/layer5io/meshery-adapter-library/api/grpc"
+	configprovider "github.com/layer5io/meshery-adapter-library/config/provider"
+	internalconfig "github.com/layer5io/meshery-osm/internal/config"
```
[the remark this answers is not in the document]
Comment (Aisuko) [reply main.go:22]: Nice catch @mgfeller  Do not sure why the `internal/config` was added to `.gitignore`, and correct this.

File: go.mod (hunk 3 above)
[the remark this answers is not in the document]
Comment (Aisuko) [reply go.mod]: Thanks

**Review by kumarabd — APPROVED** [review]

**Comment by kumarabd** [comment]
@mgfeller good to go?

File: internal/config/operations.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+func getOperations(op adapter.Operations) adapter.Operations {
+	op[OSMOperation] = &adapter.Operation{
+		Type:        int32(meshes.OpCategory_INSTALL),
+		Description: "OSM Service Mesh",
+		Versions:    []adapter.Version{"v0.3.0", "v0.2.0"},
+		Templates:   adapter.NoneTemplate,
+		AdditionalProperties: map[string]string{
+			ServiceName: OSMOperation,
```
Comment (mgfeller) [inline internal/config/operations.go:34]: @kumarabd should this be one in Services field?

Comment (mgfeller) [reply internal/config/operations.go:34]: @kumarabd still not  quite sure what we mean by 'Service' here. It's not used in the code, as far as I can see.
My understanding is/was that this is the name of the service used to retrieve endpoint information for?

**Review by mgfeller — APPROVED** [review]
Comment / question about purpose and usage of ServiceName in operation def, otherwise LGTM.

File: internal/config/operations.go (hunk 6 above)
Comment (kumarabd) [reply internal/config/operations.go]: @mgfeller you're right. Its not being used anywhere, I just removed it.

**Pull request merged**